CDCgov/prime-reportstream
# [16141] Implement ReceiverEnrichment Function

This PR introduces a new step into the ReportStream pipeline.  Receiver Enrichment relocates the enrichment schema functionality from FHIRTranslator to the new FHIRReceiverEnrichment step.

Test Steps:
1. Observe the transforms in **resources/metadata/fhir_transforms/receivers/fhir-message-header-sample.yml**.
2. Run the pipeline locally (`../gradlew quickRun`)
3. Start with a clean environment (`../gradlew clearDB`, `../gradlew reloadSettings`, `../gradlew reloadTables`).
4. Choose a message from the Postman collection (e.g., _Single HL7 Full ELR_) and post it to the running application.
5. In the **Azure Storage Explorer** find the resulting file in the blob container and download it for inspection.
6. Search the file for the values from step 1.

## Changes
- You can view all the file changes [here](https://github.com/CDCgov/prime-reportstream/pull/16854/files).
- Changes include modifications to existing code to move the receiver enrichment code out of the translate step of the pipeline to a new step, schema changes to the database (which must be included with this PR in order for jOOQ to be able to build DAO classes from the schema), and updates to documentation.

## Checklist

### Testing

### Process

## Linked Issues
- No linked issues.

## To Be Done
- No TBD items.

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - No new application endpoints, but a new Azure function
        - elr-fhir-receiver-enrichment
- Does this PR include changes in authentication and/or authorization of existing endpoints?
    - No.
- Does this change introduce new dependencies that need vetting?
    - No.
- Does this change require changes to our infrastructure?
    - No.
- Does logging contain sensitive data?
    - No.
- Does this PR include or remove any sensitive information itself?
    - No.
